x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/AlreadyProcessedFollowingEngineException.java

```diff
@@ -9,8 +9,23 @@
 import org.elasticsearch.index.engine.VersionConflictEngineException;
 import org.elasticsearch.index.shard.ShardId;
 
+import java.util.OptionalLong;
+
 public final class AlreadyProcessedFollowingEngineException extends VersionConflictEngineException {
-    AlreadyProcessedFollowingEngineException(ShardId shardId, long seqNo) {
-        super(shardId, "operation [{}] was processed before", null, seqNo);
+    private final long seqNo;
+    private final OptionalLong existingPrimaryTerm;
+
+    AlreadyProcessedFollowingEngineException(ShardId shardId, long seqNo, OptionalLong existingPrimaryTerm) {
```

Comment:
can you add java docs to indicate when one can expect a primary term to be present?